src/Symfony/Component/Mailer/Bridge/Azure/Transport/AzureApiTransport.php

```diff
@@ -0,0 +1,336 @@
+
+        $requestOptions = [
+            'body' => json_encode($messagePayload),
+            'headers' => $headers,
+        ];
+
+        $response = $this->client->request('POST', 'https://'.$endpointURL, $requestOptions);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new HttpTransportException('Could not reach the remote Azure server.', $response, 0, $e);
+        }
+
+        if (202 !== $statusCode) {
+            try {
+                $result = $response->toArray(false);
+                throw new HttpTransportException('Unable to send an email (.'.$result['error']['code'].'): '.$result['error']['message'], $response, $statusCode);
+            } catch (DecodingExceptionInterface $e) {
+                throw new HttpTransportException('Unable to send an email: '.$response->getContent(false).sprintf(' (code %d).', $statusCode), $response, 0, $e);
+            }
+        }
+
+        $sentMessage->setMessageId(json_decode($response->getContent(false), true)['id']);
+
+        return $response;
+    }
+
+    /**
+     * Get the message request body.
+     */
+    private function getPayload(Email $email, Envelope $envelope): array
+    {
+        $addressStringifier = function (Address $address) {
+            $stringified = ['address' => $address->getAddress()];
+
+            if ($address->getName()) {
+                $stringified['displayName'] = $address->getName();
+            }
+
+            return $stringified;
+        };
+
+        $data = [
+            'content' => [
+                'html' => $email->getHtmlBody(),
+                'plainText' => $email->getTextBody(),
+                'subject' => $email->getSubject(),
+            ],
+            'recipients' => [
+                'to' => array_map($addressStringifier, $this->getRecipients($email, $envelope)),
+            ],
+            'senderAddress' => $envelope->getSender()->getAddress(),
+            'attachments' => $this->getMessageAttachments($email),
+            'userEngagementTrackingDisabled' => $this->getDisableUserEngagementTracking(),
+            'headers' => empty($headers = $this->getMessageCustomHeaders($email)) ? null : $headers,
+            'importance' => $this->getPriorityLevel($email->getPriority()),
+        ];
+
+        if ($emails = array_map($addressStringifier, $email->getCc())) {
+            $data['recipients']['cc'] = $emails;
+        }
+
+        if ($emails = array_map($addressStringifier, $email->getBcc())) {
+            $data['recipients']['bcc'] = $emails;
+        }
+
+        if ($emails = array_map($addressStringifier, $email->getReplyTo())) {
+            $data['replyTo'] = $emails;
+        }
+
+        return $data;
+    }
+
+    /**
+     * List of attachments. Please note that we limit the total size of
+     * an email request (which includes attachments) to 10MB.
```

Comment:
```suggestion
     * List of attachments.
     * Please note that we limit the total size of an email request (which includes attachments) to 10MB.
```

By "we" you mean you as someone who implements this bridge, or is it a Microsoft limit?

Reply:
It was my mistake, I just copied the description from the Azure documentation. Yes, it is a limitation of the service. (https://learn.microsoft.com/en-us/rest/api/communication/dataplane/email/send?view=rest-communication-dataplane-2023-03-31&tabs=HTTP#emailmessage)